src/components/preferences/PreferencesTab.js

```diff
@@ -0,0 +1,152 @@
+import React from "react";
+import styles from "./styles";
+import { makeStyles } from "@material-ui/core/styles";
+import Switch from "@material-ui/core/Switch";
+import Typography from "@material-ui/core/Typography";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import Divider from "@material-ui/core/Divider";
+import { Field } from "formik";
+
+const useStyles = makeStyles(styles);
+
+function browseFiles() {
+    console.log("User clicked the file browser.");
+    //TODO figure out what happens here
+    //DATA VIEW IN A DRAWER?
+}
+
+export default function PreferencesTab(props) {
+    const classes = useStyles();
+
+    const onSwitchChange = (setFieldValue, fieldName) => (event, checked) => {
+        console.log(fieldName);
+        setFieldValue(fieldName, checked);
+    };
+
+    const FormSwitch = ({
+        field: { value, onChange, ...field },
+        form: { setFieldValue },
+        ...custom
+    }) => (
+        <Switch
+            checked={!!value}
+            onChange={onSwitchChange(setFieldValue, field.name)}
+            {...custom}
+        />
+    );
+
+    return (
+        <div className={classes.preferences}>
+            <Typography className={classes.sectionHeader}>General</Typography>
+            <br />
+            <Typography className={classes.section} display="inline">
+                Remember last file path for Apps
+            </Typography>
+            <Field
+                component={FormSwitch}
+                name="preferences.rememberLastPath"
+                value="rememberLastPath"
```

Comment:
I might be remembering wrong, but I don't think you need these `value` props on your Fields?  The value for each Field should automatically be coming from your `initialValues` and mapped by the `name` prop.